src/Compilers/CSharp/Portable/Symbols/Source/SourceNamedTypeSymbol_Bases.cs

```diff
@@ -185,9 +185,14 @@ protected override void CheckInterfaces(DiagnosticBag diagnostics)
 
                             Debug.Assert(!other.Equals(@interface, TypeCompareKind.ConsiderEverything));
 
+                            if (!other.Equals(@interface, TypeCompareKind.CLRSignatureCompareOptions | TypeCompareKind.IgnoreNullableModifiersForReferenceTypes))
```

Comment:
>CLRSignatureCompareOptions [](start = 74, length = 26)

Is equality with CLRSignatureCompareOptions  checked elsewhere? Consider adding a comment about where it is handled.

Reply:
Actually this entire ```if``` looks suspicious to me. Is it ever entered? It looks like ```InterfacesAndTheirBaseInterfacesNoUseSiteDiagnostics``` is built in a way that only interfaces equal based on CLRSignatureCompareOptions are added to the set. Should we replace the ```if``` with an assert?

---
In reply to: [506609960](https://github.com/dotnet/roslyn/pull/48349#discussion_r506609960) [](ancestors = 506609960)